sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatViewModel.kt

```diff
@@ -782,4 +795,47 @@ abstract class ChatViewModel<ARGS: NavArgs>(
             }
         }
     }
+
+    suspend fun processMemberRequest(
+        contactId: ContactId,
+        messageId: MessageId,
+        type: MessageType,
+    ) {
+        viewModelScope.launch(mainImmediate) {
+
+            val errorMessage = if (type.isMemberApprove()) {
+                app.getString(R.string.failed_to_approve_member)
+            } else {
+                app.getString(R.string.failed_to_reject_member)
+            }
+
+            if (type.isMemberApprove() || type.isMemberReject()) {
+                when(messageRepository.processMemberRequest(contactId, messageId, type)) {
+                    is LoadResponse.Loading -> {}
+                    is Response.Success -> {}
+
+                    is Response.Error -> {
+                        submitSideEffect(ChatSideEffect.Notify(errorMessage))
+                    }
+                }
+            }
+        }.join()
+    }
+
+    suspend fun deleteTribe() {
+        viewModelScope.launch(mainImmediate) {
+            val chat = getChat()
+
+            if (chat.type.isTribe()) {
+                when (chatRepository.exitAndDeleteTribe(chat)) {
+                    is Response.Error -> {
+                        submitSideEffect(ChatSideEffect.Notify(app.getString(R.string.failed_to_delete_tribe)))
+                    }
+                    is Response.Success -> {
+                        chatNavigator.popBackStack()
+                    }
+                }
+            }
+        }.join()
+    }
```

Comment:
These should be `open` functions that are only implemented in the `TribeChatViewModel`